src/Analytics.js

```diff
@@ -0,0 +1,202 @@
+				this.trackQuarters(2, percentagePlayed);
+			} else if (roundUpPercentage === 75 && !percentageTracked.Q3) {
+				this.trackQuarters(3, percentagePlayed);
+			} else if (roundUpPercentage === 100 && !percentageTracked.Q4) {
+				this.trackQuarters(4, percentagePlayed);
+			}
+
+			// Fire the call back of the onProgress
+			this.props.onProgress && this.props.onProgress(event);
+		}
+
+
+		/**
+		 * Method that tracks back to the "onEvent" call back to communicate a tracking of a quarter
+		 * @param {number} mark - The number of the quarter (1,2,3,4)
+		 * @param {number} percentagePlayed - The percentage played at the time of the tracking
+		 */
+		trackQuarters(mark, percentagePlayed) {
+			this.setState((prevState) => ({
+				percentageTracked: {
+					...prevState.percentageTracked,
+					[`Q${mark}`]: true
+				}
+			}));
+
+			this.onEvent({
+				type: PlayerEventTypes[`ON_PROGRESS_Q${mark}`],
+				percentagePlayed
+			});
+		}
+
+		/**
+		 * Event triggered when the fullscreen happens
+		 * @param {NativeEvent} event
+		 */
+		onEnterFullscreen(event) {
+			this.onEvent({'type': PlayerEventTypes.ON_ENTER_FULLSCREEN});
+			this.props.onEnterFullscreen && this.props.onEnterFullscreen(event);
+		}
+
+		/**
+		 * Event triggered when the user exists from the fullscreen
+		 * @param {NativeEvent} event
+		 */
+		onExitFullscreen(event) {
+			this.onEvent({'type': PlayerEventTypes.ON_EXIT_FULLSCREEN});
+			this.props.onExitFullscreen && this.props.onExitFullscreen(event);
+		}
+
+		/**
+		 * Event triggered when the user exists from the fullscreen
+		 * @param {NativeEvent} event
+		 */
+		onError(event = {}) {
+			this.onEvent({
+				'type': PlayerEventTypes.ON_EXIT_FULLSCREEN,
+				...event
+			});
+			this.props.onError && this.props.onError(event);
+		}
+
+		/**
+		 * Handler to normalise the events and send it back to the onEvent callback if that exists
+		 * @param {object} event
+		 * @param {string} event.type - the event type
+		 */
+		onEvent(event) {
+			event = {
+				...event,
+				name: this.state.mediainfo && this.state.mediainfo.title || 'N/A',
+				videoId: this.props.videoId,
+				referenceId: this.props.referenceId,
+				accountId: this.props.accountId,
+				playerId: this.player.props.playerId,
+				platform: Platform.OS
+			}
+			this.props.onEvent && this.props.onEvent(event);
+		}
+
+		render() {
+			// ... and renders the wrapped component with the fresh data!
+			// Notice that we pass through any additional props
+			return <BCPlayerComponent
+				{...this.props}
+				ref={(player) => this.player = player}
+				onReady={this.onReady.bind(this)}
+				onPlay={this.onPlay.bind(this)}
+				onPause={this.onPause.bind(this)}
+				onEnd={this.onEnd.bind(this)}
+				onProgress={this.onProgress.bind(this)}
+				onEnterFullscreen={this.onEnterFullscreen.bind(this)}
+				onExitFullscreen={this.onExitFullscreen.bind(this)}
+				onError={this.onError.bind(this)}
```

Comment:
When is this going to get called?

Reply:
@pankaj-pp  when we implement the integration of the onError with the native part. Should come in a follow up request soon.